sycl/test-e2e/ProgramManager/multi_device_bundle/device_libs_and_caching.cpp

```diff
@@ -0,0 +1,156 @@
+// REQUIRES: gpu && linux && (opencl || level_zero)
+
+// Test to check several use cases for multi-device kernel bundles.
+// Test covers AOT and JIT cases. Kernel is using some math functions to enforce
+// using device libraries to excersise additional logic in the program manager.
+// Checks are used to test that program and device libraries caching works as
+// expected.
+
+// Test JIT first.
+// Intentionally use jit linking of device libraries to check that program
+// manager can handle this as well. With this option program manager will
+// compile the main program, load and compile device libraries and then link
+// everything together.
+// RUN: %{build} -fsycl-device-lib-jit-link -o %t.out
+
+// Check the default case when in-memory caching is enabled.
+// RUN: env NEOReadDebugKeys=1 CreateMultipleRootDevices=4 SYCL_UR_TRACE=2 %{run} %t.out | FileCheck %s --check-prefixes=CHECK-SPIRV-JIT-LINK-TRACE
+
+// Check the case when in-memory caching of the programs is disabled.
+// RUN: env SYCL_CACHE_IN_MEM=0 NEOReadDebugKeys=1 CreateMultipleRootDevices=4 %{run} %t.out
+
+// Test AOT next.
+// RUN: %{build} -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device *" -o %t.out
```

Comment:
@againull, `ocloc` requirement is missing for this test, making it fail in some environments

Reply:
Thank you, missed that, will fix the test.